pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -340,7 +354,7 @@ func (a *AmbientIndexImpl) extractWorkload(p *v1.Pod, c *Controller) *model.Work
 
 	policies := c.selectorAuthorizationPolicies(p.Namespace, p.Labels)
 	policies = append(policies, c.convertedSelectorPeerAuthentications(p.Namespace, p.Labels)...)
-	wl := c.constructWorkload(p, waypoint, policies)
+	wl := c.constructWorkload(p, waypoint, policies, a.servicesMap)
```

Comment:
nit: we could just move `constructWorkload` to `ambientIndex` and we don't need to pass `a.servicesMap` since it can access it directly?

Reply:
addressed in https://github.com/istio/istio/pull/45621/commits/99b26b2759c3ddd417fc27149ac284fb2b8dfd48